packages/access-client/src/agent.js

```diff
@@ -404,6 +421,43 @@ export class Agent {
     }
   }
 
+  /**
+   * Request authorization of a session allowing this agent to issue UCANs
+   * signed by the passed email address.
+   *
+   * @param {string} email
+   * @param {object} [opts]
+   * @param {AbortSignal} [opts.signal]
+   */
+  async authorize(email, opts) {
+    const parts = email.split('@').map(s => encodeURIComponent(s))
+    const sessionPrincipal = DID.parse(`did:mailto:${parts[1]}:${parts[0]}`)
+
+    const res = await Access.authorize.invoke({
+      issuer: this.issuer,
+      audience: this.connection.id,
+      with: this.issuer.did(),
+      nb: { as: sessionPrincipal.did() }
+    }).execute(this.connection)
+
+    if (res?.error) {
+      throw new Error('failed to authorize session', { cause: res })
+    }
+
+    const sessionDelegation =
+      /** @type {Ucanto.Delegation<[import('./types').AccessSession]>} */
+      (await this.#waitForDelegation(opts))
+
+    const cap = sessionDelegation.capabilities.find(c => c.can === Access.session.can && c.nb.key === this.issuer.did())
+    if (!cap && isExpired(sessionDelegation)) {
```

Comment:
```suggestion
    if (!cap || isExpired(sessionDelegation)) {
```

Reply:
I'm worried if I commit this, it will lead to new errors when claiming expired delegations.
might wanna do this first https://github.com/web3-storage/w3protocol/issues/526